### PR TITLE
Split scanner to Solidity and Yul kinds

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: liblangutil/Scanner.cpp [hunk 1]
```diff
@@ -950,7 +959,12 @@ tuple<Token, unsigned, unsigned> Scanner::scanIdentifierOrKeyword()
 	while (isIdentifierPart(m_char) || (m_char == '.' && m_supportPeriodInIdentifier))
 		addLiteralCharAndAdvance();
 	literal.complete();
-	return TokenTraits::fromIdentifierOrKeyword(m_tokens[NextNext].literal);
+	auto const token = TokenTraits::fromIdentifierOrKeyword(m_tokens[NextNext].literal);
+	if (m_kind == ScannerKind::Yul) {
```
Comment (axic) [inline liblangutil/Scanner.cpp]: Likewise could turn off other non-Yul token in other parts of the scanner (like the various symbols).

File: liblangutil/Scanner.cpp [hunk 2]
```diff
@@ -941,6 +941,15 @@ Token Scanner::scanNumber(char _charSeen)
 	return Token::Number;
 }
 
+namespace
+{
+    constexpr bool isYulToken(Token tok)
```
Comment (axic) [inline liblangutil/Scanner.cpp]: This would need to move to `Token.h`.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `c84a9946c2e1196ac0cbbc619f7a7df9b15c72e3`:
```
Coding style error:
 liblangutil/Scanner.cpp:941: constexpr bool isYulToken(Token tok)
 liblangutil/Scanner.cpp:942: {
 liblangutil/Scanner.cpp:944: || tok == Token::Default || tok == Token::For || tok == Token::Break || tok == Token::Continue /*|| tok == Token::Leave*/;
 liblangutil/Scanner.cpp:945: }

```
Please check that your changes are working as intended.

File: liblangutil/Scanner.cpp [hunk 3]
```diff
@@ -941,16 +935,32 @@ Token Scanner::scanNumber(char _charSeen)
 	return Token::Number;
 }
 
+#if 0
+namespace
+{
+    constexpr bool isYulToken(Token tok)
```
Comment (stackenbotten) [inline liblangutil/Scanner.cpp]: Coding style error

File: liblangutil/Scanner.cpp [hunk 4]
```diff
@@ -941,16 +935,32 @@ Token Scanner::scanNumber(char _charSeen)
...
+    {
```
Comment (stackenbotten) [inline liblangutil/Scanner.cpp]: Coding style error

File: liblangutil/Scanner.cpp [hunk 5]
```diff
@@ -941,16 +935,32 @@ Token Scanner::scanNumber(char _charSeen)
...
+	return tok == Token::Function || tok == Token::Let || tok == Token::If || tok == Token::Switch || tok == Token::Case
+	    || tok == Token::Default || tok == Token::For || tok == Token::Break || tok == Token::Continue /*|| tok == Token::Leave*/;
```
Comment (stackenbotten) [inline liblangutil/Scanner.cpp]: Coding style error

File: liblangutil/Scanner.cpp [hunk 6]
```diff
@@ -941,16 +935,32 @@ Token Scanner::scanNumber(char _charSeen)
...
+    }
```
Comment (stackenbotten) [inline liblangutil/Scanner.cpp]: Coding style error

File: liblangutil/Scanner.cpp (hunk 3 above)
Comment (stackenbotten) [inline liblangutil/Scanner.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `3e5b6c34532306fc24722421b8c4e70ac24600d4`:
```
Coding style error:
 liblangutil/Scanner.cpp:941: constexpr bool isYulToken(Token tok)
 liblangutil/Scanner.cpp:942: {
 liblangutil/Scanner.cpp:944: || tok == Token::Default || tok == Token::For || tok == Token::Break || tok == Token::Continue /*|| tok == Token::Leave*/;
 liblangutil/Scanner.cpp:945: }

```
Please check that your changes are working as intended.

File: liblangutil/Scanner.cpp (hunk 4 above)
Comment (stackenbotten) [inline liblangutil/Scanner.cpp]: Coding style error

File: liblangutil/Scanner.cpp (hunk 5 above)
Comment (stackenbotten) [inline liblangutil/Scanner.cpp]: Coding style error

File: liblangutil/Scanner.cpp (hunk 6 above)
Comment (stackenbotten) [inline liblangutil/Scanner.cpp]: Coding style error

**Comment by axic** [comment]
Simplified this to only deal with the kind and replacing the `supportPeriodInIdentifier` function. @chriseth I think we could merge this as a simple refactoring.

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
This was actually merged into the tests PR and not into develop :)

**Pull request merged**